akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java

```
@@ -126,6 +126,20 @@ public void mustBeAbleToUseDropWhile() throws Exception {
     future.toCompletableFuture().get(3, TimeUnit.SECONDS);
   }
 
+  @Test
+  public void mustBeAbleToFlattenOptional() {
+    final Flow<Optional<Integer>, NotUsed> sample = Flow.range(1, 3).map(i -> Optional.of(i));
+    CompletionStage<Integer> future = sample.<Integer>flattenOptional()
+            .runWith(Sink.fold(0, (agg, next) -> agg + next), materializer);
+    try {
+        assertEquals(future.toCompletableFuture().get(),(Integer)6);
+    } catch (InterruptedException e) {
+        e.printStackTrace();
+    } catch (ExecutionException e) {
+        e.printStackTrace();
+    }
```

just let the test throw an exception, then it will fail